discord_bots/cogs/schedule.py

```diff
@@ -0,0 +1,348 @@
+        inputs = [x for x in inputs if x != ""]  # clean up empty inputs
+        pattern = r"^\d{1,2}:\d{2}[APap][Mm]$"
+
+        for input in inputs:
+            if not re.match(pattern, input):
+                await interaction.response.send_message(
+                    embed=Embed(
+                        description="Invalid time format.  Valid formats:\n- 7:00am\n- 2:00pm\n- 11:00AM",
+                        colour=Colour.red(),
+                    ),
+                    ephemeral=True,
+                )
+                return
+
+        schedule_channel: TextChannel = await interaction.guild.create_text_channel(
+            "schedule",
+            topic="Welcome to the scheduling channel!  Click the Time buttons to toggle your availability for a specific day and time.  Click the Add All button to add to all times for that day.",
+        )
+
+        with Session() as session:
+            session.add(DiscordChannel(name="schedule", channel_id=schedule_channel.id))
+
+            for day in DAYS:
+                embed = Embed(title=day, colour=Colour.blue())
+                message = await schedule_channel.send(embed=embed)
+                timestamp_date: date = ScheduleUtils.get_next_date(day)
+
+                for input in inputs:
+                    time_to_add: time = datetime.strptime(input, "%I:%M%p").time()
+                    timestamp_date_time: datetime = datetime.combine(
+                        timestamp_date, time_to_add
+                    )
+                    timestamp = discord.utils.format_dt(timestamp_date_time, style="t")
+                    embed.add_field(
+                        name=f"|            {timestamp}            |",
+                        value="",
+                        inline=True,
+                    )
+                    session.add(
+                        Schedule(day=day, time=time_to_add, message_id=message.id)
+                    )
+
+                try:
+                    session.commit()
+                except IntegrityError as exc:
+                    _log.error(f"integrity error {exc}")
+                    session.rollback()
+
+                await message.edit(embed=embed, view=ScheduleView(day))
+
+        await interaction.followup.send(
+            embed=Embed(description="Schedule created", colour=Colour.green())
+        )
+
+
+class ScheduleUtils:
+    @classmethod
+    async def rebuild_embed(self, guild: Guild, day: str):
+        with Session() as session:
+            schedule_channel_id = (
+                session.query(DiscordChannel.channel_id)
+                .filter(DiscordChannel.name == "schedule")
+                .scalar()
+            )
+            schedule_channel = get(guild.text_channels, id=schedule_channel_id)
+
+            schedules_for_day = ScheduleUtils.get_schedules_for_day(day)
+            message: Message = await schedule_channel.fetch_message(
```

Comment:
Since you're only editing the message after fetching it, I believe you could use: https://discordpy.readthedocs.io/en/stable/api.html#discord.TextChannel.get_partial_message

`get_partial_message` doesn't make an HTTP request to discord while `fetch_message` does. Partial messages have slightly less functionality, but in this case I don't think you need any more.